service/controller/app/v1/resource/chart/desired.go

```diff
@@ -76,6 +93,22 @@ func generateTarballURL(baseURL string, appName string, version string) (string,
 	return u.String(), nil
 }
 
+func hasConfigMap(cr v1alpha1.App, appCatalog v1alpha1.AppCatalog) bool {
+	if key.AppConfigMapName(cr) != "" || appcatalogkey.ConfigMapName(appCatalog) != "" {
+		return true
+	}
+
+	return false
+}
+
+func hasSecret(cr v1alpha1.App, appCatalog v1alpha1.AppCatalog) bool {
+	if key.AppSecretName(cr) != "" || appcatalogkey.SecretName(appCatalog) != "" {
```

Comment:
Where is the difference between app and appcatalog secret name? Why do check for both?

Reply:
The app CR supports 3 levels of config app, catalog and user. Any of the 3 can be specified. But if none are specified we won't create a secret for chart-operator.

The user config support will be added later because it isn't needed for VOO.

Reply:
When there are 3, why do we only check for two?

Reply:
Oh, because user config support comes later. With you.